stdlib/http/src/main/ballerina/src/http/websocket/websocket_failover.bal

```diff
@@ -0,0 +1,194 @@
+    }
+
+    # Gives the connection id associated with this connection.
+    #
+    # + return - the unique id associated with the connection
+    public function getConnectionId() returns string {
+        return self.id;
+    }
+
+    # Gives the subprotocol if any that is negotiated with the client.
+    #
+    # + return - The subprotocol if any negotiated with the client or `nil`
+    public function getNegotiatedSubProtocol() returns string? {
+        return self.negotiatedSubProtocol;
+    }
+
+    # Gives the secured status of the connection.
+    #
+    # + return - `true` if the connection is secure.
+    public function isSecure() returns boolean {
+        return self.secure;
+    }
+
+    # Gives the open or closed status of the connection.
+    #
+    # + return - `true` if the connection is open
+    public function isOpen() returns boolean {
+        return self.open;
+    }
+
+    # Gives the HTTP response if any received for the client handshake request.
+    #
+    # + return - the HTTP response received for the client handshake request
+    public function getHttpResponse() returns Response? {
+        return self.response;
+    }
+};
+
+# Configuration for the WebSocket client endpoint.
+#
+# + callbackService - The callback service for the client. Resources in this service gets called on receipt of messages
+#                     from the server.
+# + subProtocols - Negotiable sub protocols for the client
+# + customHeaders - Custom headers which should be sent to the server
+# + idleTimeoutInSeconds - Idle timeout of the client. Upon timeout, `onIdleTimeout` resource (if defined) in the client
+#                          service will be triggered.
+# + readyOnConnect - `true` if the client is ready to receive messages as soon as the connection is established.
+#                    This is true by default. If changed to false the function ready() of the
+#                    `WebSocketClient`needs to be called once to start receiving messages.
+# + secureSocket - SSL/TLS related options
+# + maxFrameSize - The maximum payload size of a WebSocket frame in bytes.
+#                  If this is not set or is negative  or zero the default frame size of 65536 will be used.
+# + targetUrls - The set of urls which are used to connect the server.
+# + failoverInterval - The maximum number of milliseconds to delay a failover attempt.
+# + webSocketCompressionEnabled - Enable support for compression in WebSocket
+# + retryConfig - Configurations related to retry
+public type WebSocketFailoverClientEndpointConfiguration record {|
+    service? callbackService = ();
+    string[] subProtocols = [];
+    map<string> customHeaders = {};
+    int idleTimeoutInSeconds = -1;
+    boolean readyOnConnect = true;
+    ClientSecureSocket? secureSocket = ();
+    int maxFrameSize = 0;
+    string[] targetUrls = [];
+    int failoverInterval = 1000;
+    boolean webSocketCompressionEnabled = true;
+    WebSocketRetryConfig retryConfig?;
+|};
```

Comment:
If some of these fields which are initialized to empty array/nil can be made optional?